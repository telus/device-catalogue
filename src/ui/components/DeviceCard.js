import React from "react";
import FlexGrid from "@tds/core-flex-grid";
import Heading from "@tds/core-heading";
import Card from "@tds/core-card";
import PriceLockup from "@tds/core-price-lockup";
import Text from "@tds/core-text";
import Image from "@tds/core-image";
import Button from "@tds/core-button";
import ChevronLink from "@tds/core-chevron-link";
import Box from "@tds/core-box";
import DimpleDivider from "@tds/core-dimple-divider";

export default function DeviceCardItem(props) {
  return (
    <div className="device-card_item">
      <FlexGrid outsideGutter={false}>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={9}>
            <Card fullHeight variant="defaultWithBorder">
              <Box between={4}>
                <Image src={props.url} width={200} height={100} alt="device" />
                <Heading level="h3">{props.brand.toUpperCase()}</Heading>
                <Heading level="h2" tag="h3">
                  {props.name}
                </Heading>
                <PriceLockup
                  size="medium"
                  price={props.price}
                  bottomText="TELUS Easy Payment and Bring-It-Back applied. Read legal footnote. Taxes due upfront."
                  signDirection="left"
                />
                <br />
                <Text size="medium">Retail price: $470</Text>
                {/* <Button>Add to cart</Button>
                <ChevronLink href="#">Learn more</ChevronLink> */}
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
      <DimpleDivider />
    </div>
  );
}
