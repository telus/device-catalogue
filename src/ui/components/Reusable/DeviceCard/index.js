import React from 'react';
import Heading from '@tds/core-heading';
import Image from '@tds/core-image';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Link from '@tds/core-link';
import Paragraph from '@tds/core-paragraph';
import Strong from '@tds/core-strong';
import Text from '@tds/core-text';
import HairlineDivider from '@tds/core-hairline-divider';
import { FootnoteLink } from '@tds/core-terms-and-conditions';
import PriceLockup from '@tds/core-price-lockup';
const DeviceCard = ({ img, path, text, brand, name, price}) => {
  const handleFootnoteClick = () => {
    console.log('footnote is clicked, terms and condition should open up from the bottom of the window');
  }
  return (
    <div className="device-card">
      <Card fullHeight>
        <Box between={4}>
          {path
           ? <Link href={path}>
              <div className="device-image">
                <Image 
                  src={img} 
                  width={200}
                  height={200}
                  alt="device-image"
                />
                <br/>
                {text}
              </div>
            </Link>
            :<div className="device-image">
              <Image 
                src={img} 
                width={200}
                height={200}
                alt="device-image"
              />
            </div>
          }
          {price &&
          <React.Fragment>
            <Strong id="device-brand">{brand}</Strong>
            <Heading level="h4" id="device-name" >{name}</Heading>
            <PriceLockup
              size="small"
              price={price}
              signDirection="left"
            />
            <HairlineDivider/>
            <Paragraph size="small">
              <Text>
                TELUS Easy Payment and Bring-It-Back applied.
                <FootnoteLink number={4} onClick={handleFootnoteClick} copy='en'/>
                <br/>
                Taxes due upfront.
              </Text>
            </Paragraph>
          </React.Fragment>
          }
        </Box>
      </Card>
    </div>
  )
}
export default DeviceCard;