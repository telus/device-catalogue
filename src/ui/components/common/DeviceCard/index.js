import React from "react";
import PropTypes from "prop-types";
import Card from "@tds/core-card";
import Box from "@tds/core-box";
import Heading from "@tds/core-heading";
import Image from "@tds/core-image";
import PriceLockup from "@tds/core-price-lockup";
import HairlineDivider from "@tds/core-hairline-divider";
import Paragraph from "@tds/core-paragraph";
import styled from "styled-components";
import { colorTelusPurple } from "@tds/core-colours";

const DeviceNameContainer = styled(Box)`
  align-items: flex-start;
`;

const DeviceName = styled.span`
  font-weight: 400;
  color: ${colorTelusPurple};
  font-size: 1.3rem;
  line-height: 1.29;
  letter-spacing: -0.8px;
`;

const DeviceCard = ({ device, image, imageAlt = "Device image" }) => {
  if (!device) {
    return null;
  }

  return (
    <Box vertical={3}>
      <Card variant="defaultWithBorder" spacing="narrow">
        <Box between={2}>
          <Box horizontal={5}>
            <Image src={image} alt={imageAlt} width={200} height={300} />
          </Box>
          <DeviceNameContainer between={1}>
            <Heading level="h3" tag="span">
              {device.brand}
            </Heading>
            <DeviceName>{device.name}</DeviceName>
          </DeviceNameContainer>
          <PriceLockup
            size="small"
            price={device.price.toString()}
            signDirection="left"
          />
        </Box>
        <Box vertical={4} between={4}>
          <HairlineDivider />
          <Paragraph>
            TELUS Easy Payment and Bring-It-Back applied. Read legal footnote
            Taxes due upfront.
          </Paragraph>
        </Box>
      </Card>
    </Box>
  );
};

DeviceCard.propTypes = {
  device: PropTypes.object,
  image: PropTypes.any,
  imageAlt: PropTypes.string,
};

export default DeviceCard;
