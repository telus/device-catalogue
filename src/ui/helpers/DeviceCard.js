import React from "react";
import FlexGrid from "@tds/core-flex-grid";
import Heading from "@tds/core-heading";
import Box from "@tds/core-box";
import Image from "@tds/core-image";
import Card from "@tds/core-card";
import HairlineDivider from "@tds/core-hairline-divider";
import PriceLockup from "@tds/core-price-lockup";
import Paragraph from "@tds/core-paragraph";

/**
 * A component that renders a Telus device card with image, name and product info
 * @param {Object} device
 * @param {imagefile} image
 */
const DeviceCard = ({ device, image }) => {
  return (
    <Card>
      <FlexGrid.Row horizontalAlign="center">
        <Box vertical={4}>
          <Image
            src={image}
            alt={`Image of ${device.name}`}
            height={200}
            width={160}
          />
        </Box>
      </FlexGrid.Row>
      <Box vertical={3} between={1}>
        <Heading level="h3" tag="h2">
          {" "}
          {device.brand}
        </Heading>
        <Heading level="h2" tag="h3">
          {" "}
          {device.name}
        </Heading>
      </Box>
      <Box>
        <PriceLockup
          size="medium"
          price={`${device.price}`}
          signDirection="left"
        />
      </Box>
      <Box between={3} vertical={4}>
        <HairlineDivider />
        <Paragraph>
          TELUS Easy Payment and Bring-It-Back applied. Read legal footnote
          Taxes due upfront.{" "}
        </Paragraph>
      </Box>
    </Card>
  );
};
export default DeviceCard;
