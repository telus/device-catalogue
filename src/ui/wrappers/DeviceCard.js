import React from 'react';
import Heading from '@tds/core-heading';
import Box from '@tds/core-box';
import PriceLockup from '@tds/core-price-lockup';
import Paragraph from '@tds/core-paragraph';
import Image from '@tds/core-image';
import Card from '@tds/core-card';
import HairlineDivider from '@tds/core-hairline-divider';

const DeviceCard = ({ name, brand, price, image }) => (
  <Card variant="defaultWithBorder">
    <Image src={image} width={158} height={200} alt="iPhone" />
    <Box vertical={2} style={{ textAlign: 'left' }}>
      <Heading level="h3" tag="h2">
        {brand}
      </Heading>
      <Paragraph size="large">{name}</Paragraph>
    </Box>
    <PriceLockup size="small" price={price} signDirection="left" />
    <Box vertical={4}>
      <HairlineDivider />
    </Box>
    <Paragraph size="medium">
      TELUS Easy Payment and Bring-It-Back applied. Read legal footnote Taxes
      due upfront.
    </Paragraph>
  </Card>
);

export default DeviceCard;
