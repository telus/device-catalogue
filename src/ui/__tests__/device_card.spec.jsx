import React from 'react';
import renderer from 'react-test-renderer';
import DeviceCard from '../components/Reusable/DeviceCard'

describe('Device card', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<DeviceCard img={''} brand='Apple' name='iPhone12' price={121}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with just 2 props', () => {
    const tree = renderer
      .create(<DeviceCard text={'iPhones'} img={''} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});