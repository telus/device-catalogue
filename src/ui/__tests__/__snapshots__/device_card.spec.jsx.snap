// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device card renders correctly 1`] = `
<div
  className="device-card"
>
  <div
    className="sc-dlfnuX Aazwe sc-hKgJUU itbWqN sc-hKgJUU itbWqN"
  >
    <div
      className="sc-dlfnuX gRMItJ"
    >
      <div
        className="device-image"
      >
        <img
          alt="device-image"
          className="Image__StyledImage-b16p20-0 DCldW"
          height={200}
          src=""
          width={200}
        />
      </div>
      <strong
        className="Strong__StyledStrong-sc-19dhku5-0 jfUeiS"
        id="device-brand"
      >
        Apple
      </strong>
      <h4
        className="Heading__StyledHeading-mpz92r-0 jGuBnj"
        data-testid="heading"
        id="device-name"
      >
        iPhone12
      </h4>
      <div
        className="sc-dlfnuX fzVLgT sc-kstqJO kmwZDm sc-kstqJO kmwZDm"
      >
        <div
          className="sc-dlfnuX dvsJHd"
        >
          <div
            className="PriceLockup__StyledRateTextWrapper-crmhv4-4 kCgybC"
          >
            <div
              className="sc-dlfnuX ebYjZi sc-hBEYId fkewPS sc-hBEYId fkewPS"
            >
              <div
                className="sc-dlfnuX ebYjZi"
              >
                <span
                  aria-hidden="false"
                  className="PriceLockup__StyledDollarSign-crmhv4-2 iteKpH"
                  data-testid="dollarSign"
                  size="small"
                >
                  $
                </span>
                <span
                  aria-hidden="false"
                  className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                  data-testid="priceValue"
                  size="small"
                >
                  121
                </span>
                <span
                  className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                  data-testid="priceValue"
                  size="small"
                >
                  <span
                    className="Text__StyledText-sc-1m0rr67-0 sc-fodVek eCTkyE iRdaTx"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr
        className="HairlineDivider__StyledHairlineDivider-biamlk-0 lnWyrV"
      />
      <p
        className="Paragraph__StyledParagraph-efl81j-0 dbIdbC"
        size="small"
      >
        <span
          className="Text__StyledText-sc-1m0rr67-0 ERlyH"
          size="base"
        >
          TELUS Easy Payment and Bring-It-Back applied.
          <sup>
            <button
              className="FootnoteLink__StyledFootnoteLink-pzcbe1-0 dkrTKu"
              data-tds-id="footnote-link"
              onClick={[Function]}
              type="button"
            >
              <span
                className="A11yContent__StyledA11yContent-sc-15wexwk-0 jolBxv"
              >
                Read legal footnote
              </span>
              4
            </button>
          </sup>
          <br />
          Taxes due upfront.
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Device card renders correctly with just 2 props 1`] = `
<div
  className="device-card"
>
  <div
    className="sc-dlfnuX Aazwe sc-hKgJUU itbWqN sc-hKgJUU itbWqN"
  >
    <div
      className="sc-dlfnuX gRMItJ"
    >
      <div
        className="device-image"
      >
        <img
          alt="device-image"
          className="Image__StyledImage-b16p20-0 DCldW"
          height={200}
          src=""
          width={200}
        />
      </div>
    </div>
  </div>
</div>
`;
